x-pack/legacy/plugins/rollup/__jest__/client_integration/job_create_metrics.test.js

```diff
@@ -247,12 +250,157 @@ describe('Create Rollup Job, step 5: Metrics', () => {
+      let getFieldChooserColumnForRow;
+      let getFieldListTableRows;
+
+      beforeEach(async () => {
+        httpRequestsMockHelpers.setIndexPatternValidityResponse({ numericFields, dateFields });
+        await goToStep(5);
+        await addFieldToList('numeric');
+        await addFieldToList('date');
+        find('rollupJobSelectAllMetricsPopoverButton').simulate('click');
+        ({ getSelectAllInputForRow, getFieldChooserColumnForRow, getFieldListTableRows } = metrics);
+      });
+
+      const expectAllFieldChooserInputs = (fieldChooserColumn, expected) => {
+        const inputs = fieldChooserColumn.reactWrapper.find('input');
+        inputs.forEach((input) => {
+          expect(input.props().checked).toBe(expected);
+        });
+      };
+
+      it('should select all of the fields in a row',  async () => {
+        // The last column is the eui "actions" column
+        const selectAllCheckbox = getSelectAllInputForRow(0);
+        selectAllCheckbox.simulate('change', { checked: true });
+        const fieldChooserColumn = getFieldChooserColumnForRow(0);
+        expectAllFieldChooserInputs(fieldChooserColumn, true);
+      });
+
+      it('should deselect all of the fields in a row ',  async () => {
+        const selectAllCheckbox = getSelectAllInputForRow(0);
+        selectAllCheckbox.simulate('change', { checked: true });
+
+        let fieldChooserColumn = getFieldChooserColumnForRow(0);
+        expectAllFieldChooserInputs(fieldChooserColumn, true);
+
+        selectAllCheckbox.simulate('change', { checked: false });
+        fieldChooserColumn = getFieldChooserColumnForRow(0);
+        expectAllFieldChooserInputs(fieldChooserColumn, false);
+      });
+
+      it('should select all of the metric types across rows (column-wise)',  () => {
+        const selectAllAvgCheckbox = find('rollupJobMetricsSelectAllCheckbox-avg');
+        selectAllAvgCheckbox.first().simulate('change', { checked: true });
+
+        const rows = getFieldListTableRows();
+
+        rows
+          .filter((row) => {
+            const [, metricTypeCol ] = row.columns;
+            return metricTypeCol.value === 'numeric';
+          })
+          .forEach((row, idx) => {
+            const fieldChooser = getFieldChooserColumnForRow(idx);
+            fieldChooser.reactWrapper.find('input').forEach(input => {
+              const props = input.props();
+              if (props['data-test-subj'].endsWith('avg')) {
+                expect(props.checked).toBe(true);
+              } else {
+                expect(props.checked).toBe(false);
+              }
+            });
+          });
+      });
+
+      it('should deselect all of the metric types across rows (column-wise)',  () => {
+        const selectAllAvgCheckbox = find('rollupJobMetricsSelectAllCheckbox-avg');
+        selectAllAvgCheckbox.last().simulate('change', { checked: true });
+        selectAllAvgCheckbox.last().simulate('change', { checked: false });
```

Comment:
can you explain why both of these lines are needed?

Reply:
Will do!